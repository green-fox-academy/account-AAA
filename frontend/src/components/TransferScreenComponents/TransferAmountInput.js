import React from 'react';
import {
  Item, View, Input, Text, Icon,
} from 'native-base';
import PropTypes from 'prop-types';

export default function TransferAmountInput({ transferAccountAmount, depositAmount }) {
  const [transferAmountInput, setAmount] = React.useState('');
  const handleInputChange = (text) => {
    setAmount(parseFloat(text));
    transferAccountAmount(parseFloat(text));
  };

  return (
    <View style={{ margin: 10 }}>
      <Text>How much to transfer:</Text>
      <Item style={{ margin: 10 }}>
        <Icon name="dollar" type="FontAwesome" />
        <Input
          onChangeText={(text) => handleInputChange(text)}
          keyboardType="numeric"
        />
      </Item>
      {(transferAmountInput > depositAmount)
        ? <Text style={{ color: 'red', fontSize: 15 }}>{`* Maximum transfer amount: $ ${depositAmount}`}</Text>
        : <></>}
    </View>
  );
}

TransferAmountInput.propTypes = {
  transferAccountAmount: PropTypes.func.isRequired,
  depositAmount: PropTypes.number.isRequired,
};
