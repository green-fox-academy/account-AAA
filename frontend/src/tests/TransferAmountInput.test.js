import React from 'react';
import renderer from 'react-test-renderer';
import TransferAmountInput from '../components/TransferScreenComponents/TransferAmountInput';

describe('<TransferAmountInput />', () => {
  it('render correctly', () => {
    const testForThisTest = () => null;
    const tree = renderer
      .create(
        <TransferAmountInput transferAccountAmount={testForThisTest} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
