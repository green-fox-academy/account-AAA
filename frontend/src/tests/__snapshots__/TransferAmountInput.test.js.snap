// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransferAmountInput /> render correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "margin": 10,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
  >
    How much to transfer:
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 1,
        "borderColor": "#D9D5DC",
        "borderLeftWidth": 0,
        "borderRadius": undefined,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "borderWidth": undefined,
        "flexDirection": "row",
        "margin": 10,
        "marginLeft": 2,
        "opacity": 1,
      }
    }
  >
    <Text />
    <TextInput
      allowFontScaling={true}
      editable={true}
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholderTextColor="#575757"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#000",
          "flex": 1,
          "fontSize": 17,
          "height": 50,
          "paddingLeft": 5,
          "paddingRight": 5,
          "top": 1.5,
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
    />
  </View>
</View>
`;
